### ISSUE TITLE
Results on faster res50 def.

### ISSUE THREAD

**Issue by AresGao** [unsky/Deformable-ConvNets-caffe#11]
Hi,have you test your results on faster res50 def?

**Comment by RuiminChen** [comment]
@unsky Hi, thanks for your work. I have tried training ResNet50 deformable Fast RCNN end2end model on WIDER FACE DataSet, the loss seemed normally converge, but I got nothing when I test my model. Moreover, I have tried training deformable VGG16 on Imagenet, the loss could not even converge. Do you have any idea about this?
Best


![2017-12-29 18 33 22](https://user-images.githubusercontent.com/13010590/34435204-ccf1b6f2-ecc6-11e7-9d4b-81ea1406d068.png)
![2017-12-29 18 33 05](https://user-images.githubusercontent.com/13010590/34435208-d1d12946-ecc6-11e7-9a2d-82f3d8a85047.png)

**Comment by unsky** [comment]
the setting of batch norm is very very import in detection task,you can refer the setting in my model.
ps:i think there have mang bugs in your test

**Comment by RuiminChen** [comment]
@unsky I don't think I have many bugs in my test, here is my results using your res50 faster rcnn config but replace deformable convolution by normal convolution, trained on WIDER FACE with 120000 iterations.
![2017-12-29 20 14 37](https://user-images.githubusercontent.com/13010590/34436907-f4022ab6-ecd4-11e7-8bf2-7232391e91e7.png)
And all my BatchNorm config in training is like below：
layer {
    bottom: "res3a_branch1"
    top: "res3a_branch1"
    name: "bn3a_branch1"
    type: "BatchNorm"
    batch_norm_param {
        use_global_stats: true
    }
    param {
        lr_mult: 0.0
        decay_mult: 0.0
    }
    param {
        lr_mult: 0.0
        decay_mult: 0.0
    }
    param {
        lr_mult: 0.0
        decay_mult: 0.0
    }
}

layer {
    bottom: "res3a_branch1"
    top: "res3a_branch1"
    name: "scale3a_branch1"
    type: "Scale"
    scale_param {
        bias_term: true
    }
}
Can you share your deformable resnet50 faster rcnn trained model on PASCAL VOC to help me check my mistakes？
Thank you

**Comment by unsky** [comment]
@RuiminChen  
https://github.com/unsky/Deformable-ConvNets-caffe/blob/master/deformable_conv_faster-rcnn/models/pascal_voc/ResNet50/faster_rcnn_end2end/train.prototxt
i think we should fix scale params

**Comment by lawpdas** [comment]
Hi,@RuiminChen ! Have you solved the problem on VGG16?

**Comment by RuiminChen** [comment]
@lawpdas yes, my results seems normal now on  both resnet50 and VGG16

**Comment by lawpdas** [comment]
@RuiminChen What should I do with VGG16? Can you give me some suggestions?
Thanks!

**Comment by haoliyoupai09** [comment]
@RuiminChen Could you tell me whether transferring the deformable convolutional network to the VGG 16 network works or not? I'm planing to apply the deformable convolutional network to the visual tracking field and quite curious about the effect of it.
Looking forward to your reply!

**Comment by zhanglonghao1992** [comment]
@RuiminChen  Hi ~ I have trained a classification network using deformable ResNet50 and the loss could converge. But when i viewed the offset the model have learned, I found it was very big. It seems like the network dosn't even use these deformable conv layer because the offsets are out of the image boundary..
How do you train the vgg net for ImageNet cls task?

**Comment by rog93** [comment]
@zhanglonghao1992 You may check DCNV2 here https://github.com/msracver/Deformable-ConvNets, They have already solved a possible issue when the sampling location is outside of image boundary

**Comment by Jokercy** [comment]
> 
> 
> @lawpdas yes, my results seems normal now on both resnet50 and VGG16
Could you tell me some details about how to solve the problem?

**Comment by Jokercy** [comment]
> 
> 
> @unsky I don't think I have many bugs in my test, here is my results using your res50 faster rcnn config but replace deformable convolution by normal convolution, trained on WIDER FACE with 120000 iterations.
> ![2017-12-29 20 14 37](https://user-images.githubusercontent.com/13010590/34436907-f4022ab6-ecd4-11e7-8bf2-7232391e91e7.png)
> And all my BatchNorm config in training is like below：
> layer {
> bottom: "res3a_branch1"
> top: "res3a_branch1"
> name: "bn3a_branch1"
> type: "BatchNorm"
> batch_norm_param {
> use_global_stats: true
> }
> param {
> lr_mult: 0.0
> decay_mult: 0.0
> }
> param {
> lr_mult: 0.0
> decay_mult: 0.0
> }
> param {
> lr_mult: 0.0
> decay_mult: 0.0
> }
> }
> 
> layer {
> bottom: "res3a_branch1"
> top: "res3a_branch1"
> name: "scale3a_branch1"
> type: "Scale"
> scale_param {
> bias_term: true
> }
> }
> Can you share your deformable resnet50 faster rcnn trained model on PASCAL VOC to help me check my mistakes？
> Thank you

Could you tell me some details about how to solve the problem?

**Comment by niuchuangnn** [comment]
Hi, @AresGao @RuiminChen @unsky @lawpdas @haoliyoupai09 , could you tell me if there exist bugs in the code, as in https://github.com/unsky/Deformable-ConvNets-caffe/issues/31#issuecomment-450848046? Thank you.
